libconcurrent/concurrent/simqueue.c

```diff
@@ -1,30 +1,45 @@
 #include <sim.h>
 #include <simqueue.h>
 
-static const int GUARD_VALUE = INT_MAX;
 static const int LOCAL_POOL_SIZE = _SIM_LOCAL_POOL_SIZE_;
-static EnqState *connect_state = NULL;
 
 static inline void EnqStateCopy(EnqState *dest, EnqState *src);
```

Comment:
Why do you need these 4 definitions here?

Reply:
I think it's a good programming practice to declare all the functions that are going to be used in the be beginning of the .c file.
I strictly follow this rule only for static functions.